pyravendb/store/document_session.py

```diff
@@ -19,6 +20,126 @@ def __init__(self, commands, deferred_command_count, entities=None):
         self.deferred_command_count = deferred_command_count
 
 
+class _RefEq:
+    def __init__(self, ref):
+        if isinstance(ref, _RefEq):
+            self.ref = ref.ref
+        self.ref = ref
+
+    def __eq__(self, other):
+        return (
+            True
+            if id(self.ref) == id(other)
+            else False
+            if not isinstance(other, _RefEq)
+            else id(self.ref) == id(other.ref)
+        )
+
+    def __hash__(self):
+        return id(self.ref)
+
+
+class _RefEqEntityHolder(object):
+    def __init__(self):
+        self.mutable_items = dict()
+
+    def __len__(self):
+        return len(self.mutable_items)
+
+    def __contains__(self, item):
+        return _RefEq(item) in self.mutable_items
+
+    def __delitem__(self, key):
+        del self.mutable_items[_RefEq(key)]
+
+    def __setitem__(self, key, value):
+        self.mutable_items[_RefEq(key)] = value
+
+    def __getitem__(self, key):
+        return self.mutable_items[_RefEq(key)]
+
+    def __getattribute__(self, item):
+        if item == "mutable_items":
+            return super().__getattribute__(item)
+        return self.mutable_items.__getattribute__(item)
+
+
+class _DocumentsByEntityHolder(object):
+    def __init__(self):
+        self._immutable_items = dict()
```

Comment:
actually, wasn't it hashable vs not hashable?

Reply:
changed to hashable and unhashable